src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -620,26 +694,39 @@ public CatalogObjectMetadata createCatalogObjectRevision(String bucketName, Stri
         CatalogObjectRevisionEntity revisionEntity = buildCatalogObjectRevisionEntity(commitMessage,
                                                                                       username,
                                                                                       projectName,
+                                                                                      tags,
                                                                                       rawObject,
                                                                                       catalogObject,
                                                                                       metadataListParsed);
 
         return new CatalogObjectMetadata(revisionEntity);
     }
 
-    protected String getProjectNameIfExistsOrEmptyString(List<Metadata> metadataListParsed) {
-        Optional<Metadata> projectNameIfExists = metadataListParsed.stream()
-                                                                   .filter(property -> property.getKey()
-                                                                                               .equals(PROJECT_NAME))
-                                                                   .findAny();
-        return projectNameIfExists.map(value -> value.getValue() == null ? "" : value.getValue()).orElse("");
+    protected String getTagsValueIfExistsOrEmptyString(List<Metadata> metadataListParsed, String metadataLabel) {
+        List<String> tagsValueIfExists = metadataListParsed.stream()
+                                                           .filter(property -> property.getLabel()
+                                                                                       .equals(metadataLabel))
+                                                           .collect(Collectors.toList())
+                                                           .stream()
+                                                           .map(Metadata::getKey)
+                                                           .collect(Collectors.toList());
+        return (tagsValueIfExists.isEmpty() ? "" : String.join(",", tagsValueIfExists));
```

Comment:
I think you can simplify doing directly

```
return metadataListParsed.stream()
                .filter(property -> property.getLabel()
                        .equals(metadataLabel))
                .map(Metadata::getKey)
                .collect(Collectors.joining(","));
```

` .collect(Collectors.joining(","));` returns an empty string if the stream is empty